map2loop/sample_storage.py

```diff
@@ -0,0 +1,268 @@
+
+    def type(self):
+        return self.storage_label
+
+    def set_default_samplers(self):
+        """
+        Initialisation function to set or reset the point samplers
+        """
+        self.samplers[SampleType.STRUCTURE] = SamplerDecimator(1)
+        self.samplers[SampleType.FAULT_ORIENTATION] = SamplerDecimator(1)
+        self.samplers[SampleType.GEOLOGY] = SamplerSpacing(50.0)
+        self.samplers[SampleType.FAULT] = SamplerSpacing(50.0)
+        self.samplers[SampleType.FOLD] = SamplerSpacing(50.0)
+        self.samplers[SampleType.CONTACT] = SamplerSpacing(50.0)
+        self.samplers[SampleType.DTM] = SamplerSpacing(50.0)
+        # dirty flags to false after initialisation
+        self.sampler_dirtyflags = [False] * len(SampleType)
+
+    @beartype.beartype
+    def set_sampler(self, sampletype: SampleType, sampler: Sampler):
+        """
+        Set the point sampler for a specific datatype
+
+        Args:
+            sampletype (Datatype):
+                The sample type (SampleType) to use this sampler on
+            sampler (Sampler):
+                The sampler to use
+        """
+        self.samplers[sampletype] = sampler
+        # set the dirty flag to True to indicate that the sampler has changed
+        self.sampler_dirtyflags[sampletype] = True
+
+    @beartype.beartype
+    def get_sampler(self, sampletype: SampleType):
+        """
+        Get the sampler name being used for a datatype
+
+        Args:
+            sampletype: The sample type of the sampler
+
+        Returns:
+            str: The name of the sampler being used on the specified datatype
+        """
+        return self.samplers[sampletype].sampler_label
+
+    @beartype.beartype
+    def store(self, sampletype: SampleType, data):
+        """
+        Stores the sample data.
+
+        Args:
+            sampletype (SampleType): The type of the sample.
+            data: The sample data to store.
+        """
+
+        # store the sample data
+        self.samples[sampletype] = data
+        self.sampler_dirtyflags[sampletype] = False
+
+    @beartype.beartype
+    def check_state(self, sampletype: SampleType):
+        """
+        Checks the state of the data, sample and sampler.
+
+        Args:
+            sampletype (SampleType): The type of the sample.
+        """
+
+        self.dirtyflags[StateType.DATA] = self.map_data.dirtyflags[sampletype]
+        self.dirtyflags[StateType.SAMPLER] = self.sampler_dirtyflags[sampletype]
+
+    @beartype.beartype
+    def load(self, sampletype: SampleType):
+        """
+        Loads the map data or raster map data based on the sample type.
+
+        Args:
+            sampletype (SampleType): The type of the sample.
+        """
+        datatype = Datatype(sampletype)
+
+        if datatype == Datatype.DTM:
+            self.map_data.load_raster_map_data(datatype)
+
+        else:
+            # load map data
+            self.map_data.load_map_data(datatype)
```

Comment:
I know this is from the mapdata class, but why can't we just have map_data.load(datatype)? And it takes care of what type the data is? Would probably be more future proof as raster data could also include geophysical grids